Rename RoutingSource to NavModel (#84)
bumble-tech/appyx: documentation/navmodel/index.md

# Navigation models

Navigation models are one of the most important abstractions of Appyx.

## What's a NavModel?

> Navigation models describe navigation itself – by the states and operations they define, any custom navigation mechanism can be implemented. The state is then yours to be controlled from your business logic, which then becomes the state of navigation directly.

Reviewer: ```suggestion
Navigation models describe navigation with two parameters: Routings (Navigation targets) and State (finite states for Navigation targets). You define which operations your Navigation models support and how they change the state of Routings. 
```

Author: The original phrasing has some key elements I don't want to discard. We can discuss your additions separately, documentation will need more update.

`NavModel` capabilities differ from implementation to implementation, however, typically:

> 1. They store information on the states all children 

Reviewer: ```suggestion
1. They store information of the children states
```

2. They behave like a state machine

> 3. They offer some public API to trigger changing the state of children  

Reviewer: ```suggestion
3. They offer some public API to trigger children's state change 
```
